drivers/disk/mmc_ll.h

```diff
@@ -0,0 +1,273 @@
+/*
+ * Copyright (c) 2018, ARM Limited and Contributors. All rights reserved.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+#ifndef MMC_LL_H
+#define MMC_LL_H
+
+#include <cache.h>
+#include <zephyr.h>
+
+/*
+ * Intel SoC re-use Arm Trusted Firmware (ATF) driver code in Zephyr.
+ * The migrated ATF driver code uses mmio_X macro to access the register.
+ * The following macros map mmio_X to Zephyr compatible function for
+ * register access. This allow Zephyr to re-use the ATF driver codes
+ * without massive changes.
+ */
+#define mmio_write_32(addr, data)	sys_write32((data), (addr))
+#define mmio_read_32(addr)		sys_read32((addr))
+#define mmio_setbits_32(addr, mask)	sys_set_bits((addr), (mask))
+#define mmio_clrbits_32(addr, mask)	sys_clear_bits((addr), (mask))
+
+#define udelay(usec)			k_busy_wait(usec)
+#define mdelay(msec)			k_busy_wait(msec * 1000U)
+
+#define flush_dcache_range(buf, size)	arch_dcache_range((void *)buf, size,\
+					K_CACHE_WB)
+
+#define inv_dcache_range(buf, size)	arch_dcache_range((void *)buf, size,\
+					K_CACHE_INVD)
+
+#define panic()				k_panic()
+#define zeromem(addr, size)		memset((void *)addr, 0, size)
+#define BIT_64				BIT64
+
+/* Refer to  atf/include/lib/libc/assert.h */
+#define assert(e)			__ASSERT_NO_MSG(e);
+
+/* Refer to atf/include/export/lib/utils_def_exp.h*/
+#define	U_(_x)				(_x##U)
+#define	U(_x)				U_(_x)
+
+/* Refer to atf/tools/cert_create/include/debug.h */
+#define ERROR(...)			printk("ERROR: "__VA_ARGS__)
+#define VERBOSE(...)			printk("VERBOSE: " __VA_ARGS__)
+#define WARN(...)			printk("WARNING: " __VA_ARGS__)
+#define BIT_32(nr)			(U(1) << (nr))
```

Comment:
No, this is too much. I understand the original goal of reusing the TF-A code without many changes but this is too much, you are basically re-defining 20 different functions and macros here. This is obviously not scalable because every driver or subsystem could re-define things around making harder and harder to review and maintain the code in the long run.

Reply:
Hi Carlo,

Will update the code again, and replace the macro with the exact zephyr code on the TF-A code. Thanks

Reply:
Replace most of the ATF formatted code to zephyr format, and removed the macro